tests/test_tiktoken.py

```diff
@@ -0,0 +1,139 @@
+        assert original_output == string
+        assert reloaded_wrapped_output == string
+
+    # Batched tokenization
+    wrapped_output = wrapped_tokenizer(
+        ['Hello world!', 'Hello world but longer!'])
+    original_output = original_tokenizer.encode_batch(
+        ['Hello world!', 'Hello world but longer!'])
+    reloaded_wrapped_output = reloaded_wrapped_tokenizer(
+        ['Hello world!', 'Hello world but longer!'])
+    assert wrapped_output['input_ids'] == original_output
+    assert set(wrapped_output.keys()) == {'input_ids', 'attention_mask'}
+    assert reloaded_wrapped_output == wrapped_output
+    assert wrapped_tokenizer.batch_decode(
+        wrapped_output['input_ids']) == original_tokenizer.decode_batch(
+            original_output)
+    assert reloaded_wrapped_tokenizer.batch_decode(
+        reloaded_wrapped_output['input_ids']
+    ) == original_tokenizer.decode_batch(original_output)
+
+    # With padding
+    wrapped_tokenizer.pad_token_id = wrapped_tokenizer.eos_token_id
+    reloaded_wrapped_tokenizer.pad_token_id = reloaded_wrapped_tokenizer.eos_token_id
+    wrapped_output = wrapped_tokenizer(
+        ['Hello world!', 'Hello world but longer!'], padding=True)
+    original_output = original_tokenizer.encode_batch(
+        ['Hello world!', 'Hello world but longer!'])
+    reloaded_wrapped_output = reloaded_wrapped_tokenizer(
+        ['Hello world!', 'Hello world but longer!'], padding=True)
+    for wrapped1, attn_mask, original1 in zip(wrapped_output['input_ids'],
+                                              wrapped_output['attention_mask'],
+                                              original_output):
+        original_length = len(original1)
+        assert wrapped1[:original_length] == original1
+        assert sum(attn_mask) == original_length
+
+    assert set(wrapped_output.keys()) == {'input_ids', 'attention_mask'}
+    assert reloaded_wrapped_output == wrapped_output
+
+    # Get vocab
+    wrapped_vocab = wrapped_tokenizer.get_vocab()
+    reloaded_wrapped_vocab = reloaded_wrapped_tokenizer.get_vocab()
+    assert wrapped_vocab == reloaded_wrapped_vocab
+
+    didnt_match = []
+    for key, value in wrapped_vocab.items():
+        if original_tokenizer.encode(key, allowed_special='all') == [value]:
+            continue
+        else:
+            didnt_match.append(
+                (key, original_tokenizer.encode(key,
+                                                allowed_special='all'), value))
+
+    # Decode is lossy because some bytes are not representable in utf-8
+    # see https://github.com/openai/tiktoken/blob/39f29cecdb6fc38d9a3434e5dd15e4de58cf3c80/tiktoken/core.py#L245-L247
+    # This means that the str: int vocab mapping doesn't work. Would have to look more into how other HF tokenizers handle this.
+    if model_name in ['gpt-4, gpt-3.5-turbot']:
+        assert len(didnt_match) == 77
+    elif model_name in ['text-davinci-003']:
+        assert len(didnt_match) == 14
```

Comment:
These are hardcoded to each test right? Maybe we should put them as kwargs to the test?

Reply:
yeah, let me redefine somewhere else